src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php

```diff
@@ -290,14 +342,133 @@ public function generateRobotsFileAction()
                 )
             );
 
-            return $this->redirectToRoute('admin_metas_index');
+            return $this->redirectToRoute('admin_meta_index');
         }
 
         $this->addFlash(
             'success',
             $this->trans('Successful update.', 'Admin.Notifications.Success')
         );
 
-        return $this->redirectToRoute('admin_metas_index');
+        return $this->redirectToRoute('admin_meta_index');
+    }
+
+    /**
+     * Gets form builder.
+     *
+     * @return FormBuilderInterface
+     */
+    private function getMetaFormBuilder()
+    {
+        return $this->get('prestashop.core.form.builder.meta_form_builder');
+    }
+
+    /**
+     * @return FormHandlerInterface
+     */
+    private function getMetaFormHandler()
+    {
+        return $this->get('prestashop.core.form.identifiable_object.meta_form_handler');
+    }
+
+    private function handleException(MetaException $exception)
+    {
+        if (0 !== $exception->getCode()) {
+            return $this->getExceptionByClassAndErrorCode($exception);
+        }
+
+        return $this->getExceptionByType($exception);
+    }
+
+    /**
+     * Gets exception by class and error code.
+     *
+     * @param MetaException $exception
+     *
+     * @return string
+     */
+    private function getExceptionByClassAndErrorCode(MetaException $exception)
+    {
+        $exceptionDictionary = [
+            MetaConstraintException::class => [
+                MetaConstraintException::INVALID_URL_REWRITE => $this->trans(
+                        'The %s field is not valid',
+                        'Admin.Notifications.Error',
+                        [
+                            sprintf(
```

Comment:
supposed to be used to incert double quotes in the error since the translation cant be changed - just forgot that part, but I'll include that, thanks :)